l2-contracts/contracts/bridge/L2SharedBridge.sol

```diff
@@ -42,18 +49,29 @@ contract L2ERC20Bridge is IL2Bridge, Initializable {
     /// @notice Initializes the bridge contract for later use. Expected to be used in the proxy.
     /// @param _l1Bridge The address of the L1 Bridge contract.
     /// @param _l2TokenProxyBytecodeHash The bytecode hash of the proxy for tokens deployed by the bridge.
-    /// @param _governor The address of the governor contract.
-    function initialize(address _l1Bridge, bytes32 _l2TokenProxyBytecodeHash, address _governor) external initializer {
+    /// @param _aliasedOwner The address of the governor contract.
+    function initialize(
+        address _l1Bridge,
+        bytes32 _l2TokenProxyBytecodeHash,
+        address _l2WrappedBaseTokenAddress,
+        address _l1BaseTokenAddress,
+        address _aliasedOwner
+    ) external reinitializer(2) {
         require(_l1Bridge != address(0), "bf");
         require(_l2TokenProxyBytecodeHash != bytes32(0), "df");
-        require(_governor != address(0), "sf");
+        require(_aliasedOwner != address(0), "sf");
+        require(_l2WrappedBaseTokenAddress != address(0), "L2 ShB wrong weth address");
+        require(_l1BaseTokenAddress != address(0), "L2 ShB wrong l1 base token address");
+        require(_l2TokenProxyBytecodeHash != bytes32(0), "df");
 
         l1Bridge = _l1Bridge;
+        l2WrappedBaseTokenAddress = _l2WrappedBaseTokenAddress;
+        l1BaseTokenAddress = _l1BaseTokenAddress;
 
         l2TokenProxyBytecodeHash = _l2TokenProxyBytecodeHash;
         address l2StandardToken = address(new L2StandardERC20{salt: bytes32(0)}());
         l2TokenBeacon = new UpgradeableBeacon{salt: bytes32(0)}(l2StandardToken);
```

Comment:
I dont think we should redeploy the beacon as we'll have to maintain two beacons